mir_eval/alignment.py

```diff
@@ -0,0 +1,300 @@
+
+    # Check positivity (need for correct PCS metric calculation)
+    if not np.all(reference_timestamps >= 0):
+        raise ValueError("Reference timestamps can not be below 0!")
+    if not np.all(estimated_timestamps >= 0):
+        raise ValueError("Estimate timestamps can not be below 0!")
+
+
+def ae(reference_timestamps, estimated_timestamps):
+    """Compute the absolute deviations between predicted and reference timestamps,
+    and then takes the median or average over all events
+
+    Examples
+    --------
+    >>> reference_timestamps = mir_eval.io.load_events('reference.txt')
+    >>> estimated_timestamps = mir_eval.io.load_events('estimated.txt')
+    >>> mae, aae = mir_eval.align.ae(reference_onsets, estimated_timestamps)
+
+    Parameters
+    ----------
+    reference_timestamps : np.ndarray
+        reference timestamps, in seconds
+    estimated_timestamps : np.ndarray
+        estimated timestamps, in seconds
+    Returns
+    -------
+    mae : float
+        Median absolute error
+    aae: float
+        Average absolute error
+    """
+    validate(reference_timestamps, estimated_timestamps)
+    deviations = np.abs(reference_timestamps - estimated_timestamps)
+    return np.median(deviations), np.mean(deviations)
+
+
+def pc(reference_timestamps, estimated_timestamps, window=0.3):
+    """Compute the percentage of correctly predicted timestamps. A timestamp is predicted
+    correctly if its position doesn't deviate more than the window parameter from the ground
+    truth timestamp.
+
+    Examples
+    --------
+    >>> reference_timestamps = mir_eval.io.load_events('reference.txt')
+    >>> estimated_timestamps = mir_eval.io.load_events('estimated.txt')
+    >>> pc = mir_eval.align.pc(reference_onsets, estimated_timestamps, window=0.2)
+
+    Parameters
+    ----------
+    reference_timestamps : np.ndarray
+        reference timestamps, in seconds
+    estimated_timestamps : np.ndarray
+        estimated timestamps, in seconds
+    window : float
+        Window size, in seconds
+        (Default value = .1)
+    Returns
+    -------
+    pc : float
+        Percentage of correct timestamps
+    """
+    validate(reference_timestamps, estimated_timestamps)
+    deviations = np.abs(reference_timestamps - estimated_timestamps)
+    return np.mean(deviations <= window)
+
+
+def pcs(reference_timestamps, estimated_timestamps, duration: float):
+    """Constructs segments out of predicted and estimated timestamps separately
+    out of each given timestamp vector with entries (t1,t2, ... tN), yielding segments with
+    the following (start, end) boundaries: (0, t1), (t1, t2), ... (tN, duration).
+    The metric then calculates the percentage of overlap between correct segments compared to the
+    total duration.
+
+    This metric is based on the paper
+    "LyricSynchronizer: Automatic synchronization system between musical audio signals and lyrics"
```

Comment:
Please add a Sphinx ref here (like https://github.com/craffel/mir_eval/blob/master/mir_eval/separation.py#L366)

Reply:
Ping on this one.